File: res/js/DubheStandard.js
/* eslint-disable  */

class ActionType {
    static CATEGORY = 'ACTION_TYPE';
    static CATEGORY_NAME = '操作类型';

    constructor(itemName, itemKey, itemValue, itemSeq) {
        this.itemName = itemName;
        this.itemKey = itemKey;
        this.itemValue = itemValue;
        this.itemSeq = itemSeq;
    }

    getItemName() {
        return this.itemName;
    }

    getItemValue() {
        return this.itemValue;
    }

    getItemKey() {
        return this.itemKey;
    }

    getItemSeq() {
        return this.itemSeq;
    }

    getCategory() {
        return ActionType.CATEGORY;
    }

    getCategoryName() {
        return ActionType.CATEGORY_NAME;
    }
}
ActionType.CREATE = new ActionType('创建', 'CREATE', '10', 1);  
ActionType.UPDATE = new ActionType('编辑', 'UPDATE', '20', 2);  
ActionType.REMOVE = new ActionType('删除', 'REMOVE', '30', 3);  
ActionType.QUERY = new ActionType('查询', 'QUERY', '40', 4);  
ActionType.REPORT = new ActionType('上报', 'REPORT', '50', 5);  

/************************** ActionType define end ***************************/

class YesNo {
    static CATEGORY = 'YES_NO';
    static CATEGORY_NAME = '是否';

    constructor(itemName, itemKey, itemValue, itemSeq) {
        this.itemName = itemName;
        this.itemKey = itemKey;
        this.itemValue = itemValue;
        this.itemSeq = itemSeq;
    }

    getItemName() {
        return this.itemName;
    }

    getItemValue() {
        return this.itemValue;
    }

    getItemKey() {
        return this.itemKey;
    }

    getItemSeq() {
        return this.itemSeq;
    }

    getCategory() {
        return YesNo.CATEGORY;
    }

    getCategoryName() {
        return YesNo.CATEGORY_NAME;
    }
}
YesNo.YES = new YesNo('是', 'YES', '1', 1);  
YesNo.NO = new YesNo('否', 'NO', '0', 0);  

/************************** YesNo define end ***************************/


const DubheStandard = {};

ActionType.$ALL = [

ActionType.CREATE,
ActionType.UPDATE,
ActionType.REMOVE,
ActionType.QUERY,
ActionType.REPORT,
];

Object.freeze(ActionType);
export { ActionType };

/************************** ActionType export end ***************************/

YesNo.$ALL = [

YesNo.YES,
YesNo.NO,
];

Object.freeze(YesNo);
export { YesNo };

/************************** YesNo export end ***************************/


DubheStandard['ActionType'] = ActionType.$ALL;
DubheStandard['ACTION_TYPE'] = ActionType.$ALL;
DubheStandard['YesNo'] = YesNo.$ALL;
DubheStandard['YES_NO'] = YesNo.$ALL;

export default DubheStandard;
export function FindEnum(type) {
    return type.$ALL || DubheStandard[type];
}

/* eslint-enable  */

File: res/java/DubheStandard.java
package com.kosmos.dubhe;

import com.kosmos.dubhe.standard.EnumType;
import static com.kosmos.dubhe.standard.EnumTypeUtils.ENUM_SELF_INTEGER;
import static com.kosmos.dubhe.standard.EnumTypeUtils.ENUM_SELF_STRING;

public enum DubheStandard {
    INSTANCE;

        public enum ActionType implements EnumType<ActionType> {
                CREATE("创建","CREATE","10",1),   
                UPDATE("编辑","UPDATE","20",2),   
                REMOVE("删除","REMOVE","30",3),   
                QUERY("查询","QUERY","40",4),   
                REPORT("上报","REPORT","50",5),   
            SELF(ENUM_SELF_STRING, ENUM_SELF_STRING, ENUM_SELF_STRING, ENUM_SELF_INTEGER);

            public static final String CATEGORY = "ACTION_TYPE";
            public static final String CATEGORY_NAME = "操作类型";
            public static final ActionType DEFAULT = values()[0];

            ActionType(String itemName, String itemKey, String itemValue, Integer itemSeq) {
                this.itemName = itemName;
                this.itemKey = itemKey;
                this.itemValue = itemValue;
                this.itemSeq = itemSeq;
            }

            private final String itemName;
            private final String itemKey;
            private final String itemValue;
            private final Integer itemSeq;

            public String getItemName() {
                return itemName;
            }

            public String getItemKey() {
                return itemKey;
            }

            public String getItemValue() {
                return itemValue;
            }

            public Integer getItemSeq() {
                return itemSeq;
            }

            public String getString() {
                return toString().toLowerCase();
            }

            @Override
            public  ActionType[] allValues() {
                return values();
            }

            @Override
            public String getCategory() {
                return CATEGORY;
            }

            @Override
            public String getCategoryName() {
                return CATEGORY_NAME;
            }
        }

        public enum YesNo implements EnumType<YesNo> {
                YES("是","YES","1",1),   
                NO("否","NO","0",0),   
            SELF(ENUM_SELF_STRING, ENUM_SELF_STRING, ENUM_SELF_STRING, ENUM_SELF_INTEGER);

            public static final String CATEGORY = "YES_NO";
            public static final String CATEGORY_NAME = "是否";
            public static final YesNo DEFAULT = values()[0];

            YesNo(String itemName, String itemKey, String itemValue, Integer itemSeq) {
                this.itemName = itemName;
                this.itemKey = itemKey;
                this.itemValue = itemValue;
                this.itemSeq = itemSeq;
            }

            private final String itemName;
            private final String itemKey;
            private final String itemValue;
            private final Integer itemSeq;

            public String getItemName() {
                return itemName;
            }

            public String getItemKey() {
                return itemKey;
            }

            public String getItemValue() {
                return itemValue;
            }

            public Integer getItemSeq() {
                return itemSeq;
            }

            public String getString() {
                return toString().toLowerCase();
            }

            @Override
            public  YesNo[] allValues() {
                return values();
            }

            @Override
            public String getCategory() {
                return CATEGORY;
            }

            @Override
            public String getCategoryName() {
                return CATEGORY_NAME;
            }
        }

}

